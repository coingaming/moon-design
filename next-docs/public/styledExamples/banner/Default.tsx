import { Banner } from '@heathmont/moon-core';

const Example = () => (
  <Banner title="Title here" description="Lorem ipsum dolor..." />
);

export default Example;
