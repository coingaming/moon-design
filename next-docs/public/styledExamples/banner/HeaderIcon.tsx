import { Banner } from '@heathmont/moon-core';
import { ControlsClose } from '@heathmont/moon-icons';

const Example = () => (
  <Banner
    title="Title here"
    description="Lorem ipsum dolor..."
    iconHeader={<ControlsClose />}
  />
);

export default Example;
