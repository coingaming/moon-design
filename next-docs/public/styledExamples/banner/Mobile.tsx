import { Banner, Button } from '@heathmont/moon-core';

const Example = () => {
  const buttons = [
    <Button variant="secondary">Discard</Button>,
    <Button>Confirm</Button>,
  ];
  return (
    <Banner
      title="Title here"
      description="Lorem ipsum dolor..."
      buttons={buttons}
      isMobile
    />
  );
};

export default Example;
