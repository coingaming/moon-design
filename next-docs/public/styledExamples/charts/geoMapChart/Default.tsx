// @ts-nocheck
import React from 'react';
import { Filter, GeoMapChart } from '@heathmont/moon-charts';
import { SingleItemSelect } from '@heathmont/moon-components';

const Example = () => {
  const data = [
    { label: 'Estonia', code: 'EST', value: 120 },
    { label: 'Russia', code: 'RUS', value: 100 },
    { label: 'Belarus', code: 'BLR', value: 80 },
    { label: 'Argentina', code: 'ARG', value: 60 },
    { label: 'USA', code: 'USA', value: 40 },
  ];

  const filter = (
    <Filter title="by GGR">
      {() => (
        <SingleItemSelect
          onChange={() => {}}
          value="ggr"
          options={[
            { label: 'Active players', value: 'actives' },
            { label: 'GGR', value: 'ggr' },
            { label: 'Bets', value: 'bets' },
          ]}
        />
      )}
    </Filter>
  );

  return (
    <GeoMapChart
      title="Top Countries"
      data={data}
      filter={filter}
      hasUpdates={true}
      onUpdate={() => {}}
      onShare={() => {}}
      onExpand={() => {}}
    />
  );
};

export default Example;
