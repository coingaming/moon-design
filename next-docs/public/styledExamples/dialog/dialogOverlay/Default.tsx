import React from 'react';
import { DialogOverlay } from '@heathmont/moon-components';
import { Button } from '@heathmont/moon-core';

const Example = () => {
  const [showDialog, setShowDialog] = React.useState(false);

  return (
    <>
      <Button variant="tertiary" onClick={() => setShowDialog(true)}>
        Open Dialog
      </Button>
      <DialogOverlay
        isOpen={showDialog}
        onDismiss={() => setShowDialog(false)}
      />
    </>
  );
};

export default Example;
