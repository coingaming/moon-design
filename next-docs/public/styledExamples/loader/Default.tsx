import React from 'react';
import { Loader } from '@heathmont/moon-core';

const Example = () => <Loader />;

export default Example;
