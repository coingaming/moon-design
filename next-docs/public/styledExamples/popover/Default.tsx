import React, { useCallback, useEffect, useState } from 'react';
import { Popover } from '@heathmont/moon-popover';
import { Footer, Heading, Text, Button } from '@heathmont/moon-core';

const Example = () => {
  const [open, setOpen] = useState(false);

  const HandlerOpen = useCallback(
    (e: any) => {
      e.preventDefault();
      setOpen(false);
    },
    [open]
  );

  return (
    <Popover
      padding={0}
      placement="bottom"
      isOpen={open}
      onVisibilityChange={(status) => {
        setOpen(status);
        console.log('status: ', status);
      }}
      content={
        <div>
          <div className="p-8 space-y-4">
            <Heading size={20}>Agents</Heading>
            <Text size={14}>
              Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do
              eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut
              enim ad minim veniam, quis nostrud exercitation ullamco laboris
              nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in
              reprehenderit in voluptate velit esse cillum dolore eu fugiat
              nulla pariatur.
            </Text>
          </div>

          <Footer
            primButton={
              <Button fullWidth size="small" onClick={HandlerOpen}>
                OK
              </Button>
            }
            secButton={
              <Button
                fullWidth
                variant="secondary"
                size="small"
                onClick={HandlerOpen}
              >
                Cansel
              </Button>
            }
            tertButton={
              <Button
                fullWidth
                variant="tertiary"
                size="small"
                onClick={HandlerOpen}
              >
                Decline
              </Button>
            }
            isDivider
            size="small"
          />
        </div>
      }
    >
      <Button>Toggle Popover</Button>
    </Popover>
  );
};

export default Example;
