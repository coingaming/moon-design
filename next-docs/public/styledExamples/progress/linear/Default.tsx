import React from 'react';
import { ProgressLinear } from '@heathmont/moon-components';

const Example = () => (
  <>
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
    />
    <ProgressLinear
      value={30}
      text={<span>Progress</span>}
      backgroundColor="hit.100"
      height={5}
    />
  </>
);

export default Example;
