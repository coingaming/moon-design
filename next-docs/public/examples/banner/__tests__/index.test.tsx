/**
 * @jest-environment jsdom
 */

import React from 'react';
import { moonDesignLight, ThemeProvider } from '@heathmont/moon-themes';
import renderer from 'react-test-renderer';
import Buttons from '../Buttons';
import Default from '../Default';
import HeaderIcon from '../HeaderIcon';
import Mobile from '../Mobile';
import Singleline from '../Singleline';

const renderWithTheme = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>{component}</ThemeProvider>
);

const renderWithThemeRTL = (component: JSX.Element) => (
  <ThemeProvider theme={moonDesignLight}>
    <div dir="rtl">{component}</div>
  </ThemeProvider>
);

describe('Banner', () => {
  it('renders', () => {
    const testRenderer = renderer.create(renderWithTheme(<Default />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with buttons', () => {
    const testRenderer = renderer.create(renderWithTheme(<Buttons />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with buttons RTL', () => {
    const testRenderer = renderer.create(renderWithThemeRTL(<Buttons />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders with header icon', () => {
    const testRenderer = renderer.create(renderWithTheme(<HeaderIcon />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders in mobile mode', () => {
    const testRenderer = renderer.create(renderWithTheme(<Mobile />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });

  it('renders in single line mode', () => {
    const testRenderer = renderer.create(renderWithTheme(<Singleline />));
    expect(testRenderer.toJSON()).toMatchSnapshot();
  });
});
