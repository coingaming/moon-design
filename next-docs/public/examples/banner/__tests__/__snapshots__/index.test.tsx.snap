// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner renders 1`] = `
<div
  className="sc-dAoLme jxgugp"
>
  <div
    className="sc-inlEHy itKAVX"
  >
    <div
      className="sc-fusXOv fgEDRg"
    >
      <div
        className="sc-xIxFg gIRSJd"
      >
        Title here
      </div>
    </div>
    <div
      className="sc-bSQsih fLlkTC"
    >
      Lorem ipsum dolor...
    </div>
  </div>
  <div
    className="sc-lohPeM bLuILi"
  />
</div>
`;

exports[`Banner renders in mobile mode 1`] = `
<div
  className="sc-dAoLme cyTBSC"
>
  <div
    className="sc-inlEHy itKAVX"
  >
    <div
      className="sc-fusXOv fgEDRg"
    >
      <div
        className="sc-xIxFg gIRSJd"
      >
        Title here
      </div>
    </div>
    <div
      className="sc-bSQsih fLlkTC"
    >
      Lorem ipsum dolor...
    </div>
  </div>
  <div
    className="sc-lohPeM bLuILi"
  >
    <span>
      <button
        className="sc-jxUbYT cmTSxM"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="medium"
      >
        Discard
        <div
          className="sc-dZldlu fkiYmq"
        />
      </button>
    </span>
    <span>
      <button
        className="sc-jxUbYT dazUMb"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="medium"
      >
        Confirm
        <div
          className="sc-dZldlu fkiYmq"
        />
      </button>
    </span>
  </div>
</div>
`;

exports[`Banner renders in single line mode 1`] = `
<div
  className="sc-dAoLme jxgugp"
>
  <div
    className="sc-inlEHy RVLAS"
  >
    <div
      className="sc-bSQsih PfGxY"
    >
      Your email is not verified. Verify to enable withdrawals.
    </div>
    <div
      className="sc-lohPeM bLuILi"
    >
      <button
        className="sc-jxUbYT eiHueD"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="medium"
      >
        Resend verification link to email â€º
        <div
          className="sc-dZldlu fkiYmq"
        />
      </button>
    </div>
  </div>
</div>
`;

exports[`Banner renders with buttons 1`] = `
<div
  className="sc-dAoLme bXteL"
>
  <div
    className="sc-inlEHy itKAVX"
  >
    <div
      className="sc-fusXOv fgEDRg"
    >
      <div
        className="sc-xIxFg gIRSJd"
      >
        Title here
      </div>
    </div>
    <div
      className="sc-bSQsih fLlkTC"
    >
      Lorem ipsum dolor...
    </div>
  </div>
  <div
    className="sc-lohPeM bLuILi"
  >
    <span>
      <button
        className="sc-jxUbYT cmTSxM"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="medium"
      >
        Discard
        <div
          className="sc-dZldlu fkiYmq"
        />
      </button>
    </span>
    <span>
      <button
        className="sc-jxUbYT dazUMb"
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        size="medium"
      >
        Confirm
        <div
          className="sc-dZldlu fkiYmq"
        />
      </button>
    </span>
  </div>
</div>
`;

exports[`Banner renders with buttons RTL 1`] = `
<div
  dir="rtl"
>
  <div
    className="sc-dAoLme bXteL"
  >
    <div
      className="sc-inlEHy itKAVX"
    >
      <div
        className="sc-fusXOv fgEDRg"
      >
        <div
          className="sc-xIxFg gIRSJd"
        >
          Title here
        </div>
      </div>
      <div
        className="sc-bSQsih fLlkTC"
      >
        Lorem ipsum dolor...
      </div>
    </div>
    <div
      className="sc-lohPeM bLuILi"
    >
      <span>
        <button
          className="sc-jxUbYT cmTSxM"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          size="medium"
        >
          Discard
          <div
            className="sc-dZldlu fkiYmq"
          />
        </button>
      </span>
      <span>
        <button
          className="sc-jxUbYT dazUMb"
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          size="medium"
        >
          Confirm
          <div
            className="sc-dZldlu fkiYmq"
          />
        </button>
      </span>
    </div>
  </div>
</div>
`;

exports[`Banner renders with header icon 1`] = `
<div
  className="sc-dAoLme jxgugp"
>
  <div
    className="sc-inlEHy itKAVX"
  >
    <div
      className="sc-fusXOv fgEDRg"
    >
      <div
        className="sc-xIxFg gIRSJd"
      >
        Title here
      </div>
      <div
        className="sc-crMKhx logKHT"
      >
        <svg
          className="sc-JZRVG"
          fill="none"
          height="1em"
          viewBox="0 0 32 32"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M7 7l9 9m0 0l-9 9m9-9l9 9m-9-9l9-9"
            stroke="currentColor"
            strokeLinecap="round"
            strokeLinejoin="round"
          />
        </svg>
      </div>
    </div>
    <div
      className="sc-bSQsih fLlkTC"
    >
      Lorem ipsum dolor...
    </div>
  </div>
</div>
`;
